content/en/blog/_posts/2024-04-17-kubernetes-1.30.md

```diff
@@ -0,0 +1,218 @@
+
+The Kubernetes repo now uses Go workspaces. This should not impact end users at all, but does have a impact for developers of downstream projects. Switching to workspaces caused some breaking changes in the flags to the various [k8s.io/code-generator](https://github.com/kubernetes/code-generator) tools. Downstream consumers should look at [`staging/src/k8s.io/code-generator/kube_codegen.sh`](https://github.com/kubernetes/code-generator/blob/master/kube_codegen.sh) to see the changes.
+
+For full details on the changes and reasons why Go workspaces was introduced, read [Using Go workspaces in Kubernetes](https://www.kubernetes.dev/blog/2024/03/19/go-workspaces-in-kubernetes/).
+
+## Improvements that graduated to beta in Kubernetes v1.30 {#graduations-to-beta}
+
+_This is a selection of some of the improvements that are now beta following the v1.30 release._
+
+### Node log query ([SIG Windows](https://github.com/kubernetes/community/tree/master/sig-windows))
+
+To help with debugging issues on nodes, Kubernetes v1.27 introduced a feature that allows viewing logs of services running on the node. To use the feature, ensure that the NodeLogQuery feature gate is enabled for that node, and that the kubelet configuration options enableSystemLogHandler and enableSystemLogQuery are both set to true. 
+
+On Linux the assumption is that service logs are available via journald. On Windows the assumption is that service logs are available in the application log provider. On both operating systems, logs are also available by reading files within /var/log/.
+For more information, see the [log query](/docs/concepts/cluster-administration/system-logs/#log-query) documentation.
+
+### CRD Validation Ratcheting ([SIG API Machinery](https://github.com/kubernetes/community/tree/master/sig-api-machinery))
+
+You need to enable the `CRDValidationRatcheting`
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) to
+use this behavior, which then applies to all CustomResourceDefinitions in your
+cluster.
+
+Provided you enabled the feature gate, Kubernetes implements _validation racheting_
+for CustomResourceDefinitions. The API server is willing accept updates to resources that
+are not valid after the update, provided that each part of the resource that failed to validate
+was not changed by the update operation. In other words, any invalid part of the resource
+that remains invalid must have already been wrong.
+You cannot use this mechanism to update a valid resource so that it becomes invalid.
+
+This feature allows authors of CRDs to confidently add new validations to the
+OpenAPIV3 schema under certain conditions. Users can update to the new schema
+safely without bumping the version of the object or breaking workflows.
+
+### Contextual logging ([SIG Instrumentation](https://github.com/kubernetes/community/tree/master/sig-instrumentation))
+
+Blurb for 3077 goes here.
```

Comment:
```suggestion
Contextual Logging advances to beta in Kubernetes 1.30, empowering developers and operators to inject customizable, correlatable contextual details like service names and transaction IDs into logs through `WithValues` and `WithName`. This enhancement simplifies the correlation and analysis of log data across distributed systems, significantly improving the efficiency of troubleshooting efforts. By offering a clearer insight into the workings of your Kubernetes environments, Contextual Logging ensures that operational challenges are more manageable, marking a notable step forward in Kubernetes observability.
```

Reply:
OK, but: please wrap the Markdown at around 100 characters.

Reply:
@sftim Kristin will do an overall pass to wrap all lines as discussed in our release comms meeting recently